resources/assets/js/features/search/components/Result/Result.js

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+import Filter from './Filter';
+import SearchForm from './SearchForm';
+import TripList from './TripList';
+import SortPanel from './SortPanel';
+import Pagination from './Pagination';
+import { connect } from 'react-redux';
+import { search, encodeCoord, getDataFromQuery, setUrl, setFilter} from '../../services/SearchService';
+import { searchResult } from '../../actions';
+import { bindActionCreators } from 'redux';
+import { withRouter, browserHistory } from 'react-router';
+import '../../styles/search-result.scss';
+
+class Result extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            collection: [],
+            meta: {
+                totalSize: 0,
+                priceRange: [0,0]
+            },
+            preloader: true,
+            sort: props.location.query.sort || 'price',
+            order: props.location.query.order || 'asc',
+            page: +props.location.query.page || 1,
+            limit: 10,
+            filter: {},
+            resetFilter: false,
+            errors: {}
+        };
+
+        this.onChangeSort = this.onChangeSort.bind(this);
+        this.onChangePage = this.onChangePage.bind(this);
+        this.onChangeFilter = this.onChangeFilter.bind(this);
+    }
+
+    componentWillMount() {
```

Comment:
strange method and business logic in it, it seems that it can be replaced with this.getData call, ensure that there is no any over complicated logic for data fetch